pkg/registry/core/service/rest.go

```diff
@@ -257,19 +257,15 @@ func (rs *REST) healthCheckNodePortUpdate(oldService, service *api.Service) (boo
 	// Insert health check node port into the service's HealthCheckNodePort field if needed.
 	case !neededHealthCheckNodePort && needsHealthCheckNodePort:
 		glog.Infof("Transition to LoadBalancer type service with ExternalTrafficPolicy=Local")
-		if err := rs.allocateHealthCheckNodePort(service); err != nil {
+		if err := rs.allocateHealthCheckNodePort(service, nodePortOp); err != nil {
 			return false, errors.NewInternalError(err)
 		}
 
 	// Case 2: Transition from needs HealthCheckNodePort to don't need HealthCheckNodePort.
 	// Free the existing healthCheckNodePort and clear the HealthCheckNodePort field.
 	case neededHealthCheckNodePort && !needsHealthCheckNodePort:
 		glog.Infof("Transition to non LoadBalancer type service or LoadBalancer type service with ExternalTrafficPolicy=Global")
-		err := rs.serviceNodePorts.Release(int(oldHealthCheckNodePort))
-		if err != nil {
-			glog.Warningf("error releasing service health check %s node port %d: %v", service.Name, oldHealthCheckNodePort, err)
-			return false, errors.NewInternalError(fmt.Errorf("failed to free health check nodePort: %v", err))
-		}
+		nodePortOp.ReleaseDeferred(int(oldHealthCheckNodePort))
 		glog.Infof("Freed health check nodePort: %d", oldHealthCheckNodePort)
```

Comment:
Yeah that sentence may not be accurate anymore, and V(0) seems too verbose. Revised a bit and increased the verbose level.